coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/RelToTrinoConverter.java

```diff
@@ -130,7 +122,28 @@ public Result visit(Uncollect e) {
     // Build <unnestColumns>
     final List<SqlNode> unnestOperands = new ArrayList<>();
     for (RexNode unnestCol : ((Project) e.getInput()).getChildExps()) {
-      unnestOperands.add(x.qualifiedContext().toSql(null, unnestCol));
+      if (e instanceof HiveUncollect && unnestCol.getType().getSqlTypeName().equals(SqlTypeName.ARRAY)
+          && unnestCol.getType().getComponentType().getSqlTypeName().equals(SqlTypeName.ROW)) {
```

Comment:
Could you please add an example of the intended transformation as a comment before the `if` statement? Might as well update the RB description.

Reply:
Added both